node_modules/oae-activity/lib/test/util.js

```diff
@@ -440,3 +440,56 @@ var getFullySetupPushClient = module.exports.getFullySetupPushClient = function(
         });
     });
 };
+
+/**
+ * Wait for a message to come through the activity that matches the given criteria
+ *
+ * @param  {PushClient}     client              The WS client on which to listen for messages
+ * @param  {String}         activityType        The type of activity to wait for
+ * @param  {String}         verb                The verb of the activity to wait for
+ * @param  {String}         actorId             The id of the actor who performed the activity to wait for
+ * @param  {String}         [objectId]          The id of the object of the activity to wait for. If unspecified, indicates that you don't care what the object id is
+ * @param  {String}         [targetId]          The id of the target of the activity to wait for. If unspecified, indicates that you don't care what the target id is
+ * @param  {Function}       callback            Invoked when a message with the specified activity has been received
+ * @param  {Activity}       callback.activity   The activity object that was contained in the message
+ */
+var waitForPushActivity = module.exports.waitForPushActivity = function(client, activityType, verb, actorId, objectId, targetId, callback) {
+
+    /*!
+     * Listener function to wait for messages, perform the activity filter, and unbind itself from
+     * the client when the activity has been found. When found, the callback will be invoked with
+     * the activity
+     */
+    var _onMessage = function(message) {
+        var targetActivity = _.chain(message.activities)
+            .filter(function(activity) {
+                if (activity[ActivityConstants.properties.OAE_ACTIVITY_TYPE] !== activityType) {
+                    return false;
+                } else if (activity.verb !== verb) {
+                    return false;
+                } else if (activity.actor[ActivityConstants.properties.OAE_ID] !== actorId) {
+                    return false;
+                } else if (objectId && activity.object[ActivityConstants.properties.OAE_ID] !== objectId) {
+                    return false;
+                } else if (targetId) {
+                    if (!activity.target) {
+                        return false;
+                    } else if (activity.target[ActivityConstants.properties.OAE_ID] !== targetId) {
+                        return false;
+                    }
+                }
+
+                return true;
+            })
+            .first()
```

Comment:
Since we're doing `.first()`, would we not be able to replace this with:
```javascript
var targetActivity = _.find(message.activities, function(activity) {
                if (activity[ActivityConstants.properties.OAE_ACTIVITY_TYPE] !== activityType) {
                    return false;
                } else if (activity.verb !== verb) {
                    return false;
                } else if (activity.actor[ActivityConstants.properties.OAE_ID] !== actorId) {
                    return false;
                } else if (objectId && activity.object[ActivityConstants.properties.OAE_ID] !== objectId) {
                    return false;
                } else if (targetId) {
                    if (!activity.target) {
                        return false;
                    } else if (activity.target[ActivityConstants.properties.OAE_ID] !== targetId) {
                        return false;
                    }
                }

                return true; 
});
```
?